fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/TextIndexTest.java

```diff
@@ -1039,7 +1043,167 @@ public void saveComplexWithAggressiveConflictRanges() throws Exception {
     }
 
     @Test
-    public void tokenizerVersionChange() throws Exception {
+    void deleteWhereSimpleDocuments() {
+        final SimpleDocument doc1 = SimpleDocument.newBuilder()
+                .setDocId(1623L)
+                .setGroup(0)
+                .setText("foo bar")
+                .build();
+        final SimpleDocument doc2 = SimpleDocument.newBuilder()
+                .setDocId(1945L)
+                .setGroup(0)
+                .setText("foo bar")
+                .build();
+        final RecordMetaDataHook hook = metaDataBuilder -> {
+            TextIndexTestUtils.addRecordTypePrefix(metaDataBuilder);
+            // Add "text" to the simple document primary key so that when we issue a delete where based
+            // on the first field of the SimpleDocument$text index, it will align with the primary key
+            metaDataBuilder.getRecordType(SIMPLE_DOC)
+                    .setPrimaryKey(concat(recordType(), field("text"), field("doc_id")));
+        };
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+            recordStore.saveRecord(doc1);
+            recordStore.saveRecord(doc2);
+            commit(context);
+        }
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+            Query.InvalidExpressionException err = assertThrows(Query.InvalidExpressionException.class,
+                    () -> recordStore.deleteRecordsWhere(SIMPLE_DOC, Query.field("text").equalsValue("foo bar")));
```

Comment:
Why doesn't this work? `SimpleDocument$text` is a value index on `text`, so clearing the range on `"foo bar"` should work, right?